### PR TITLE
Skip flakey mobile library test crashes

### PR DESCRIPTION
**Pull request by mdh1418**
The work done in #50800 and #51491 addressed the majority of test failures and crashes in Android, iOS, tvOS, and MacCatalyst. Addressing flakey tests failures and flakey test suite crashes (where the suite crashes in CI but runs and passes locally) is an ongoing process.

In [this dropbox paper](https://www.dropbox.com/scl/fi/5pacol17z9yube0l95vib/Monitoring-Runtime-Staging-Health.paper?dl=0&rlkey=u9asjcqgdmq69jx16cx2qu5ms), many lanes on the `runtime-staging` pipeline are monitored to see which test suites fail. 

This PR adds the failing test suites to test exclusions in `tests.proj` and will fall under the issue https://github.com/dotnet/runtime/issues/52615.

### CODE REVIEW

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -118,11 +118,35 @@
...
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Xml/tests/Misc/System.Xml.Misc.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/System.Xml.XmlNodeReader.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Xml/tests/XmlSerializer/System.Xml.XmlSerializer.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Xml/tests/XPath/XPathDocument/System.Xml.XPath.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Uri/tests/ExtendedFunctionalTests/System.Private.Uri.ExtendedFunctional.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Reflection.DispatchProxy/tests/System.Reflection.DispatchProxy.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Reflection.Emit/tests/System.Reflection.Emit.Tests.csproj" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetOS)' == 'Android' and '$(TargetArchitecture)' == 'x64'">
```
Comment (akoeplinger) [inline src/libraries/tests.proj:144]: nit: this and other Android ItemGroups should have the `RunDisabledAndroidTests` condition so we can still run them (btw. do we have a draft PR which runs all the tests somewhere?)

**Review by akoeplinger — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

File: src/libraries/tests.proj (hunk 1 above)
Comment (mdh1418) [reply src/libraries/tests.proj:144]: Good catch! I'll put up a draft PR to run all of these tests as well.

Comment (steveisok) [reply src/libraries/tests.proj:144]: We'll need to bump this https://github.com/dotnet/runtime/pull/50095

**Review by akoeplinger — APPROVED** [review]

File: src/libraries/tests.proj [hunk 2]
```diff
@@ -159,13 +199,18 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Net.Security/tests/FunctionalTests/System.Net.Security.Tests.csproj" />
 
     <!-- Crashes on CI but passes locally https://github.com/dotnet/runtime/issues/52615 -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Microsoft.Extensions.Http.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)Microsoft.Extensions.Logging/tests/Common/Microsoft.Extensions.Logging.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.IO.Compression/tests/System.IO.Compression.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.IO.Compression.Brotli/tests/System.IO.Compression.Brotli.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.IO.Compression.Brotli/tests/System.IO.Compression.ZipFile.Tests.csproj" />
```
Comment (akoeplinger) [inline src/libraries/tests.proj:206]: The path is wrong here, should be System.IO.Compression.ZipFile.Tests/tests, though I just checked our analytics and don't see any failures of Brotli or ZipFile on Android in the last 7 days, so maybe we don't need this?

Comment (mdh1418) [reply src/libraries/tests.proj]: Good catch :) Thanks!

**Comment by mdh1418** [comment]
The test suites failing on `runtime-staging` both here and in #50095 are all related to `Package Installation Error`

**Pull request merged**